internal/query/user.go

```diff
@@ -347,6 +341,27 @@ var (
 //go:embed user_by_id.sql
 var userByIDQuery string
 
+func userCheckPermission(ctx context.Context, resourceOwner string, userID string, permissionCheck domain.PermissionCheck) bool {
+	ctxData := authz.GetCtxData(ctx)
+	if ctxData.UserID != userID {
+		if err := permissionCheck(ctx, domain.PermissionUserRead, resourceOwner, userID); err != nil {
+			return false
+		}
+	}
+	return true
+}
+
+func (q *Queries) GetUserByIDWithPermission(ctx context.Context, shouldTriggerBulk bool, userID string, permissionCheck domain.PermissionCheck) (*User, error) {
+	user, err := q.GetUserByID(ctx, shouldTriggerBulk, userID)
+	if err != nil {
+		return nil, err
+	}
+	if !userCheckPermission(ctx, user.ResourceOwner, user.ID, permissionCheck) {
+		return nil, zerrors.ThrowPermissionDenied(nil, "QUERY-RUz27qGst0", "Errors.PermissionDenied")
+	}
+	return user, nil
+}
```

Comment:
wouldn't it make sense to return the error of the permission check instead of having to create one again where needed
if only a boolean is needed (list checks) we can easily check on the error != nil

Reply:
Changed to error handling